lib/engine/game/g_18_ms.rb

```diff
@@ -42,6 +74,32 @@ def operating_round(round_num)
           [Step::BuyCompany, blocks: true],
         ], round_num: round_num)
       end
+
+      def or_set_finished
+        case @turn
+        when 3 then rust('2+', 20)
+        when 4 then rust('3+', 30)
+        when 5 then rust('4+', 60)
+        end
+      end
+
+      private
+
+      def rust(train, salvage_value)
+        rusted_trains = []
+        trains.each do |t|
```

Comment:
Would be tempted to do this as a rusted_trains = trains.select

Then do the rust later as a rusted_trains.each

Reply:
Fixed